### PR TITLE
Add RSS to TorrentProject

### PR DESCRIPTION
**Pull request by duramato**

### CODE REVIEW

**Comment by SiCKRAGEB0T** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

**Comment by fernandog** [comment]
Dont have size, seeders, leechers in RSS?

**Comment by duramato** [comment]
@fernandog it does

**Comment by fernandog** [comment]
So you need to add the "discarding" log message and the check about minseeders/minleeders

**Comment by duramato** [comment]
@fernandog example?

**Comment by fernandog** [comment]
https://github.com/duramato/SickRage/blob/develop/sickbeard/providers/alpharatio.py#L140-L144

**Comment by duramato** [comment]
@fernandog done

**Comment by fernandog** [comment]
Import helpers
seeders = helpers.tryInt(item['seeders'], 0)
leechers = helpers.tryInt(item['leechers'], 0)

File: sickbeard/providers/torrentproject.py [hunk 1]
```diff
@@ -64,9 +69,8 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
                 results = []
                 for i in torrents:
                     title = torrents[i]["title"]
-                    seeders = int(torrents[i]["seeds"])
-                    leechers = int(torrents[i]["leechs"])
-
+                    seeders = helpers.tryInt(torrents[i]["seeds"],1)
```
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: always put a space after a comma...

File: sickbeard/providers/torrentproject.py [hunk 2]
```diff
@@ -52,7 +53,11 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
                 if mode != 'RSS':
                     logger.log(u"Search string: %s " % search_string, logger.DEBUG)
 
-                searchURL = self.urls['api'] + "?s=%s&out=json&filter=2101" % quote_plus(search_string.encode('utf-8'))
+                if mode == 'RSS':
```
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: this change is unnecessary, change the search string in the rss call

File: sickbeard/providers/torrentproject.py [hunk 3]
```diff
@@ -108,13 +116,13 @@ def __init__(self, provider_obj):
...
+        randomtime = random.randrange(10, 25)
+        self.minTime = randomtime
 
     def _getRSSData(self):
-        # no rss for torrentproject afaik,& can't search with empty string
-        # newest results are always > 1 day since added anyways
-        # search_strings = {'RSS': ['']}
-        return {'entries': {}}
+        
+        search_params = {'RSS': ['']}
```
Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: `search_params = {'RSS': ['0day']}`

Comment (duramato) [inline sickbeard/providers/torrentproject.py]: but it also needs the &num=150 afaik

Comment (miigotu) [inline sickbeard/providers/torrentproject.py]: You can put num=150 in both RSS and non RSS, no need to selectively build the url and complicate the code.

**Pull request merged**